packages/react-utilities/src/compose/README.md

```diff
@@ -74,120 +82,62 @@ const ToggleButton = React.forwardRef((props, ref) => {
   const state = useToggleButton(props, ref);
 
   // Inject classNames as needed.
-  state.className = css(
-    state.className,
-    styles.root,
-    state.checked && styles.checked
-  );
+  state.className = mergeClasses(state.className, styles.root, state.checked && styles.checked);
 
   return renderButton(state);
+});
 ```
 
 ### Details
 
-#### Creating mutable state with `mergeProps`
-
-In the previous example, `_.merge` was used to deep clone the props into a state object. Creating a single clone and
-using that to construct state simplifies hook development and usage; rather than trying to re-clone objects
-unnecessarily on every small mutation, hooks can assume operating against a draft state. This creates more self
-contained hooks, which can ensure they apply state updates correctly, avoiding accidents like stomping on existing event
-handlers by blind object assigning the results.
-
-However, deep merge overlooks many edge cases for component props:
-
-- Deep merging classnames should append them, not replace
-- Deep merging JSX, ref objects, or arrays should replace, not recurse/clone
-- Deep merging an object on a string should replace
-
-...which introduces the first utility: `mergeProps`. Merge props works like a deep merge, but takes care of classnames,
-JSX, arrays, and object edge cases.
```

Comment:
We don't use deep merge anymore, see RFC #18642. I removed this section completely as it does not make sense.